lib/system.nim

```diff
@@ -2137,16 +2139,12 @@ when notJSnotNims:
 
 when not defined(js):
   proc cmp(x, y: string): int =
-    when defined(nimscript):
+    when nimvm:
       if x < y: result = -1
       elif x > y: result = 1
       else: result = 0
     else:
-      when nimvm:
-        if x < y: result = -1
-        elif x > y: result = 1
-        else: result = 0
-      else:
+      when not defined(nimscript):
```

Comment:
`when not defined(nimscript):` reduces to `when true` because nimscript always runs in nimvm

Reply:
Yes but it semantic checks the else branch. We dealt with this already. If you change this to `when true` you get `undeclared identifier nimCmpMem`